crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs

```diff
@@ -240,16 +240,16 @@ impl<'a> SideEffectDetector<'a> {
           || self.detect_side_effect_of_expr(&expr.right)
       }
 
-      // TODO: Implement these
       Expression::Super(_)
       | Expression::AwaitExpression(_)
       | Expression::ChainExpression(_)
       | Expression::ImportExpression(_)
       | Expression::TaggedTemplateExpression(_)
       | Expression::UpdateExpression(_)
       | Expression::YieldExpression(_)
-      | Expression::JSXElement(_)
-      | Expression::JSXFragment(_) => true,
+
+      // TODO: Implement these
+      | Expression::JSXElement(_) | Expression::JSXFragment(_) => true,
```

Comment:
Please recover the todo message, and mark these two as `unreachable`, we can assume that in `side_effects_detect phase`, ast has been transpiled. And as for above you still need to analyze them case by case.

Reply:
Should the `.with_jsx(true)` [here](https://github.com/rolldown/rolldown/blob/3fe402c075b4d7937b229bc6f71de243efded376/crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs#L459) also be updated to `.with_jsx(false)`?

For the branches above JSX:
- Super: This cannot be used alone, and all the usage has side effects.
- AwaitExpression: Both rollup and esbuild regards this has-side-effect/not-removable. However, I am unsure about awaiting known primitive types like `await 1`.
- ChainExpression: Fixed.
- ImportExpression: Always have side effects.
- TaggedTemplateExpression: `#PURE` notation not works both in rollup and esbuild.
- UpdateExpression: Always have side effects.
- YieldExpression: Always have side effects. (except `yield* []`? But this is too rare)

Reply:
for, `TaggedTemplateExpression`: 
https://github.com/evanw/esbuild/blob/9c13ae1f06dfa909eb4a53882e3b7e4216a503fe/internal/js_ast/js_ast_helpers.go#L2578-L2590